openedx/core/djangoapps/programs/tasks/v1/tests/test_tasks.py

```diff
@@ -165,12 +165,20 @@ def test_award_program_certificate(self):
         """
         test_username = 'test-username'
         test_client = EdxRestApiClient('http://test-server', jwt='test-token')
+
         httpretty.register_uri(
             httpretty.POST,
             'http://test-server/user_credentials/',
         )
+
         tasks.award_program_certificate(test_client, test_username, 123)
-        self.assertEqual(httpretty.last_request().body, json.dumps({'program_id': 123, 'username': test_username}))
+
+        expected_body = {
+            'username': test_username,
+            'credential': {'program_id': 123},
+            'attributes': []
+        }
+        self.assertEqual(httpretty.last_request().body, json.dumps(expected_body))
```

Comment:
Good question. Since dictionaries are unordered collections, I don't know how `json.dumps` could guarantee an ordering of items in its output. My guess is that this assertion just happened to work, but it definitely seems fragile. I've switched things around so that we assert that `loads(body) == expected_body`.